thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: UnresolvedTy(_#0t)'
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.